### PR TITLE
[jasmine] Make Promise more lax.

### PR DESCRIPTION
**Pull request by ienzam**
Recreates PR #33826 and
commit 3f1d1f4d8e3c2858ba7d0bd969b2172076ec511a

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by ienzam** [comment]
@chivesrs

**Review by chivesrs — APPROVED** [review]

**Comment by sandersn** [comment]
For a critical package like jest I'd like to have multiple signoffs before merging.

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -765,44 +765,44 @@ declare namespace jasmine {
          * Expect a promise to be pending, i.e. the promise is neither resolved nor rejected.
          * @param expectationFailOutput
          */
-        toBePending(expectationFailOutput?: any): Promise<void>;
+        toBePending(expectationFailOutput?: any): PromiseLike<void>;
```
Comment (sandersn) [inline types/jasmine/index.d.ts:768]: Seems like changing the return types of methods to `PromiseLike` actually makes them less easy to use, not more lax.

File: types/jasmine/tslint.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
     "extends": "dtslint/dt.json",
     "rules": {
+        "await-promise": false,
```
Comment (sandersn) [inline types/jasmine/tslint.json:4]: Why add this exception?

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/jasmine/tslint.json (hunk 2 above)
Comment (chivesrs) [reply types/jasmine/tslint.json:4]: See comment above, same deal re ts3.1/

File: types/jasmine/index.d.ts (hunk 1 above)
Comment (chivesrs) [reply types/jasmine/index.d.ts:768]: So what happened is, when the TS team made the change to remove the ts3.1/ directory and move files outward, some repositories (aka Jasmine typings) had some changes in the outer level index.d.ts that weren't reflected in ts3.1/

This is what happened here, you can see the original commit that created these typings in the original post.

I looked at the source code for Jasmine. It uses Promise.then(success, failure) callback style. I don't see Jasmine using the Promise.then(success).catch(failure) callback style at all.

I think PromiseLike is fine here.

**Review by sandersn — APPROVED** [review]
Thanks for the explanation. I was moving too fast and missed it the first time.

**Pull request merged**